server/methods/addUserToRoom.coffee

```diff
@@ -13,7 +19,7 @@ Meteor.methods
 		if room.usernames.indexOf(data.username) isnt -1
 			return
 
-		newUser = RocketChat.models.Users.findOneByUsername username: data.username
```

Comment:
You made a lot of changes.  It's hard to catch everything.  :smile: 

I like how you consolidated the logic into the models.  It should be a lot easier to maintain.